routing/turns.cpp

```diff
@@ -51,9 +51,88 @@ static_assert(g_turnNames.size() == static_cast<size_t>(TurnDirection::Count),
 
 namespace routing
 {
-namespace turns
+// UniNodeId -------------------------------------------------------------------
+bool UniNodeId::operator==(UniNodeId const & rhs) const
+{
+  if (m_type != rhs.m_type)
+  {
+    ASSERT(false, ("Comparing UniNodeId of different type. m_type", m_type, "rhs.m_type:", rhs.m_type));
+    return false;
+  }
+
+  switch (m_type) {
+  case Type::Osrm: return m_nodeId == rhs.m_nodeId;
+  case Type::Mwm: return m_featureId == rhs.m_featureId
+        && m_segId == rhs.m_segId && m_forward == rhs.m_forward;
+  }
+}
+
+bool UniNodeId::operator<(UniNodeId const & rhs) const
 {
+  if (m_type != rhs.m_type)
+  {
+    ASSERT(false, ("Comparing UniNodeId of different type. m_type", m_type, "rhs.m_type:", rhs.m_type));
```

Comment:
Up to you, but from my point of view it's not a problem. If you, say, want to store a lot of UniNodeIds in a single vector and sort them, there are two options:
1. You saving UniNodeIds of different types, and you know that.
2 .You saving UniNodeIds of different types, but you don't know that and this is actually an error.

In the case 1 the assertion you wrote here is wrong. So, if you want to compare two different UniNodeIds, why not to check that these ids are of the same time, at the call place?